plugins/inputs/opcua_listener/subscribe_client.go

```diff
@@ -133,6 +149,9 @@ func (o *SubscribeClient) Connect() error {
 
 func (o *SubscribeClient) Stop(ctx context.Context) <-chan struct{} {
 	o.Log.Debugf("Stopping OPC subscription...")
```

Comment:
Shouldn't you also reset the connection state here? Or even better, use `o.OpcUAClient.State()` to get the actual connection state of the client instead of introduce a redundant (and potential inconsistent) `successfulConnect` state here!

Reply:
That looks much better - I've changed it to use the `o.State` which is just a wrapper for `o.OpcUAClient.State()`.